session_test.go

```diff
@@ -2015,6 +2018,84 @@ func (s *testSessionSuite) TestIssue1265(c *C) {
 	mustExecFailed(c, se, "insert t values ('1e2');")
 }
 
+func (s *testSessionSuite) TestIssue1435(c *C) {
+	defer testleak.AfterTest(c)()
+	store := newStore(c, s.dbName)
+	se := newSession(c, store, s.dbName)
+	se1 := newSession(c, store, s.dbName)
+	se2 := newSession(c, store, s.dbName)
+
+	mustExecSQL(c, se, "drop table if exists t;")
+	mustExecSQL(c, se, "create table t (a int);")
+	mustExecSQL(c, se, "drop table if exists t1;")
+	mustExecSQL(c, se, "create table t1 (a int);")
+	mustExecSQL(c, se, "drop table if exists t2;")
+	mustExecSQL(c, se, "create table t2 (a int);")
+	startCh1 := make(chan struct{}, 0)
+	startCh2 := make(chan struct{}, 0)
+	endCh1 := make(chan struct{}, 0)
+	endCh2 := make(chan struct{}, 0)
+	execFailedFunc := func(c *C, s Session, tbl string, start, end chan struct{}) {
+		mustExecSQL(c, s, "begin;")
```

Comment:
If `*C` check failed in a background goroutine, the goroutine will exit immediately, test will be blocked
So we better avoid use '*C' in background goroutine.